chain-signatures/contract/tests/tests.rs

```diff
@@ -371,3 +396,144 @@ async fn test_contract_initialization() -> anyhow::Result<()> {
 
     Ok(())
 }
+
+#[tokio::test]
+async fn test_contract_propose_update() {
+    let (_, contract, accounts, _) = init_env().await;
+    dbg!(contract.id());
+
+    test_propose_update_config(&contract, &accounts).await;
+    test_propose_update_contract(&contract, &accounts).await;
+}
+
+async fn test_propose_update_config(contract: &Contract, accounts: &[Account]) {
+    // contract should not be able to propose updates unless its apart of the participant/voter set.
+    let execution = contract
+        .call("propose_update")
+        .args_json(serde_json::json!({
+            "contract": vec![1, 2, 3],
+        }))
+        .transact()
+        .await
+        .unwrap();
+    dbg!(&execution);
+    assert!(execution.is_failure());
```

Comment:
Not blocking this PR, but since we have a unified way to check if acc can vote, it would be great to check for an error type explicitly.